luigi_pipeline/lib/model/mito_mt_schema.py

```diff
@@ -0,0 +1,268 @@
+import hail as hl
+
+from lib.model.base_mt_schema import BaseMTSchema, row_annotation
+from hail_scripts.computed_fields import variant_id
+from hail_scripts.computed_fields import vep
+
+
+class SeqrMitoSchema(BaseMTSchema):
+
+    def __init__(self, *args, ref_data, high_constraint_region, **kwargs):
+        self._ref_data = ref_data
+        self._high_constraint_region = high_constraint_region
+
+        # See _selected_ref_data
+        self._selected_ref_data_cache = None
+
+        super().__init__(*args, **kwargs)
+
+    def set_mt(self, mt):
+        super().set_mt(mt)
+        # set this to None, and the @property _selected_ref_data
+        # can populate it if it gets used after each MT update.
+        self._selected_ref_data_cache = None
+
+    @property
+    def _selected_ref_data(self):
+        """
+        Reuse `self._ref_data[self.mt.row_key]` for all annotations.
+        We'll use the @property _selected_ref_data to access this
+        value, and lazily cache it (so as not to compute it if it
+        doesn't get get accessed after an update to self.mt).
+
+        Returns: self._ref_data[self.mt.row_key]
+        """
+        if not self._selected_ref_data_cache:
+            self._selected_ref_data_cache = self._ref_data[self.mt.row_key]
+        return self._selected_ref_data_cache
+
+    @row_annotation()
+    def vep(self):
```

Comment:
most of the function here look identical to those defined in ` SeqrSchema`. I think it would be better to, in the file where `SeqrSchema` is defined, split the class into `SeqrCoreSchema` which inherits from `BaseMTSchema` and has everything thats shared with mito and then have `SeqrSchema` inherit from `SeqrCoreSchema` and only have the fields not used by mito

Reply:
After studying the `BaseMTSchema` class, I found that I could remove unwanted fields from the base class by redefining the annotation function. So, I could use the `SeqrSchema` as a base class without changing it. Do you think it is a good approach?

Reply:
yes I think so